### PR TITLE
ADS-420: Add commit hooks to run prettier and linters on every commit.

### PR DESCRIPTION
**Pull request by misdreavus79**
### Description

Code editor setups can notify us of any errors (and run Prettier if configured to do so), but they don't fix any issues they find. Adding commit hooks allows us to both ensure no code is being committed with problems that can be caught, but also serves as a way to ensure we're _always_ formatting code before it makes it into prod.

### Details

I installed `lint-staged` and Husky, and configured each to run prettier on all relevant files, while running stylelint and eslint on CSS and JS files, respectively. Since it's a commit hook, no extra commands are necessary.

### To Do

In order for this commit hook to work best, we need to turn warnings into errors. The commit hook doesn't prevent you from committing code with warnings. Using the `--max-warnings=0` flag throws errors for all warnings on ESLint, but that would take us back to the original issues with the whole codebase.

### CODE REVIEW

File: src/styles/components/_accordion.scss
Comment (kaseyllabelle) [inline src/styles/components/_accordion.scss:1]: @misdreavus79 could you revert this file?

File: src/utils/__tests__/helpers.js
Comment (kaseyllabelle) [inline src/utils/__tests__/helpers.js:1]: @misdreavus79 could you revert this file?

File: package.json [hunk 1]
```diff
@@ -24,10 +24,12 @@
     "eject": "react-scripts eject",
     "generate-types": "graphql-codegen --config codegen.yml --require dotenv/config",
     "cypress:open": "cypress open",
+    "cypress:run": "cypress run",
     "sitemap": "node src/utils/generateSitemap.js",
     "stylelint:check": "stylelint-config-prettier-scss-check",
     "release-date": "echo \"// This file is automatically updated on release\nexport const releaseDate = new Date('$(date)');\" > ./src/utils/release-date.ts",
-    "lint:style": "stylelint \"**/*.scss\""
+    "lint:style": "stylelint \"**/*.scss\"",
```
Comment (APJohns) [inline package.json:31]: Could we also get a command to run eslint? Maybe for prettier too?

File: package.json [hunk 2]
```diff
@@ -24,10 +24,12 @@
     "eject": "react-scripts eject",
     "generate-types": "graphql-codegen --config codegen.yml --require dotenv/config",
     "cypress:open": "cypress open",
+    "cypress:run": "cypress run",
     "sitemap": "node src/utils/generateSitemap.js",
     "stylelint:check": "stylelint-config-prettier-scss-check",
```
Comment (APJohns) [inline package.json:29]: This script gives me an error. What does this do?

Comment (misdreavus79) [reply package.json:29]: Runs cypress in the CLI. Only necessary if we want to run cypress tests in the build pipelines.

**Review by APJohns — APPROVED** [review]

**Pull request merged**